src/integrationTest/java/uk/gov/hmcts/reform/cpo/utils/UIDService.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.cpo.utils;
+
+import org.springframework.stereotype.Service;
+
+import java.security.SecureRandom;
+
+@SuppressWarnings("PMD")
+@Service
+public class UIDService {
+
+    private final SecureRandom random = new SecureRandom();
+
+    /**
+     * Generates a random valid UID.
+     *
+     * @return A randomly generated, valid, UID.
+     */
+    public String generateUID() {
+        String currentTime10OfSeconds = String.valueOf(System.currentTimeMillis()).substring(0, 11);
+        StringBuilder builder = new StringBuilder(currentTime10OfSeconds);
+        for (int i = 0; i < 4; i++) {
+            int digit = random.nextInt(10);
+            builder.append(digit);
+        }
+        // Do the Luhn algorithm to generate the check digit.
+        int checkDigit = checkSum(builder.toString(), true);
+        builder.append(checkDigit);
+
+        return builder.toString();
```

Comment:
If we need to create a UUID only,  we could  use UUID uuid = UUID.randomUUID();
        String randomUUIDString = uuid.toString();  .  However if the business requirement is other please ignore this comments.

Reply:
I think this class is mis-named as it's actually creating valid Luhn numbers, to be used as Case Id's

For UUIDs, elsewhere in tests I do use the method you suggest